Main Class Not Found with AdoptOpneJDK/IcedTea-Web version >1.8, working in 1.8

we have an application which is started using JNLP and was working using IcedTeaWeb Version 1.8 (classpath.org, pipermail march 19). 

When I tried the same JNLP file with AdoptOpenJDK/IcedTea-Web Version 1.8.1 .. 1.8.3 I got the following errors (identical error for all three versions of IcedTea Web > 1.8)

```
selected jre: C:\Program Files\Amazon Corretto\jdk1.8.0_222\jre\
JAR https://XXXXX/YY/docs/jnlp/doc-applet.jar not found. Continuing.
JAR https://XXXXX/YY/docs/jnlp/doc-applet.jar not found. Continuing.
netx: Initialization Error: Could not initialize application. (Fatal: Initialization Error: Unknown Main-Class. Could not determine the main class for this application.)
net.sourceforge.jnlp.LaunchException: Fatal: Initialization Error: Could not initialize application. The application has not been initialized, for more information execute javaws from the command line.
        at net.sourceforge.jnlp.Launcher.createApplication(Launcher.java:822)
        at net.sourceforge.jnlp.Launcher.launchApplication(Launcher.java:531)
        at net.sourceforge.jnlp.Launcher$TgThread.run(Launcher.java:945)
Caused by: net.sourceforge.jnlp.LaunchException: Fatal: Initialization Error: Unknown Main-Class. Could not determine the main class for this application.
        at net.sourceforge.jnlp.runtime.JNLPClassLoader.initializeResources(JNLPClassLoader.java:774)
        at net.sourceforge.jnlp.runtime.JNLPClassLoader.<init>(JNLPClassLoader.java:338)
        at net.sourceforge.jnlp.runtime.JNLPClassLoader.createInstance(JNLPClassLoader.java:421)
        at net.sourceforge.jnlp.runtime.JNLPClassLoader.getInstance(JNLPClassLoader.java:495)
        at net.sourceforge.jnlp.runtime.JNLPClassLoader.getInstance(JNLPClassLoader.java:468)
        at net.sourceforge.jnlp.Launcher.createApplication(Launcher.java:814)
        ... 2 more
```
The Jar file at `https://XXXXX/YY/docs/jnlp/doc-applet.jar` is falsely not found since with 1.8 is ok ok and pointing a browser to this URL finds the JAR file. Then in the JNLP the main class is declared in the `<application-desc>`section. 

Any hints or advises would be very helpful

--- a/core/src/main/java/net/adoptopenjdk/icedteaweb/resources/downloader/JarDiffUnpacker.java
+++ b/core/src/main/java/net/adoptopenjdk/icedteaweb/resources/downloader/JarDiffUnpacker.java
@@ -1,6 +1,7 @@
 package net.adoptopenjdk.icedteaweb.resources.downloader;
 
 import net.adoptopenjdk.icedteaweb.Assert;
+import net.adoptopenjdk.icedteaweb.io.FileUtils;
 import net.adoptopenjdk.icedteaweb.io.IOUtils;
 import net.adoptopenjdk.icedteaweb.logging.Logger;
 import net.adoptopenjdk.icedteaweb.logging.LoggerFactory;
@@ -41,6 +42,13 @@ class JarDiffUnpacker implements StreamUnpacker {
 
         final JarFile originalJar = new JarFile(cacheFile);
         final File diffJarFile = new File(TMP_DIR.getFile(), UUID.randomUUID().toString() + JARDIFF_EXTENSION);
+
+        try {
+            FileUtils.createParentDir(diffJarFile);
+        } catch (IOException e) {
+            LOG.error("Error creating tmp dir for jardiff.", e);
+        }
+
         try(final FileOutputStream outputStream = new FileOutputStream(diffJarFile)) {
             IOUtils.copy(input, outputStream);
             final JarFile diffJar = new JarFile(diffJarFile);

